Zend/zend_object_handlers.c

```diff
@@ -160,27 +160,30 @@ ZEND_API HashTable *zend_std_get_debug_info(zend_object *object, int *is_temp) /
 	}
 
 	zend_call_known_instance_method_with_0_params(ce->__debugInfo, object, &retval);
-	if (Z_TYPE(retval) == IS_ARRAY) {
-		if (!Z_REFCOUNTED(retval)) {
-			*is_temp = 1;
-			return zend_array_dup(Z_ARRVAL(retval));
-		} else if (Z_REFCOUNT(retval) <= 1) {
+	switch (Z_TYPE(retval)) {
+		case IS_ARRAY:
+			if (!Z_REFCOUNTED(retval)) {
+				*is_temp = 1;
+				return zend_array_dup(Z_ARRVAL(retval));
+			} else if (Z_REFCOUNT(retval) <= 1) {
+				*is_temp = 1;
+				ht = Z_ARR(retval);
+				return ht;
+			} else {
+				*is_temp = 0;
+				zval_ptr_dtor(&retval);
+				return Z_ARRVAL(retval);
+			}
+		default:
+			zend_throw_error(NULL, ZEND_DEBUGINFO_FUNC_NAME "() must return an array");
+			zval_ptr_dtor(&retval);
+			/* fallthrough */
+		case IS_NULL:
 			*is_temp = 1;
-			ht = Z_ARR(retval);
+			ht = zend_new_array(0);
```

Comment:
Not familiar enough with this code to be sure: Does this code predate `zend_empty_array` (which would be non-reference-counted and have to have `*is_temp = 0`)

```
#define ZVAL_EMPTY_ARRAY(z) do {						\
		zval *__z = (z);								\
		Z_ARR_P(__z) = (zend_array*)&zend_empty_array;	\
		Z_TYPE_INFO_P(__z) = IS_ARRAY; \
	} while (0)
```